openstudiocore/src/bimserver/BIMserverConnection.cpp

```diff
@@ -0,0 +1,649 @@
+/**********************************************************************
+*  Copyright (c) 2008-2015, Alliance for Sustainable Energy.
+*  All rights reserved.
+*
+*  This library is free software; you can redistribute it and/or
+*  modify it under the terms of the GNU Lesser General Public
+*  License as published by the Free Software Foundation; either
+*  version 2.1 of the License, or (at your option) any later version.
+*
+*  This library is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+*  Lesser General Public License for more details.
+*
+*  You should have received a copy of the GNU Lesser General Public
+*  License along with this library; if not, write to the Free Software
+*  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+**********************************************************************/
+
+#include "BIMserverConnection.hpp"
+
+#include <QString>
+#include <QUrl>
+#include <QtNetwork/QNetworkAccessManager>
+#include <QtNetwork/QNetworkRequest>
+#include <QtNetwork/QNetworkReply>
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QJsonValue>
+#include <QByteArray>
+#include <QFileInfo>
+#include <QDebug>
+#include <iostream>
+
+
+namespace openstudio {
+namespace bimserver {
+
+  BIMserverConnection::BIMserverConnection(QObject * parent, QString bimserverAddr, QString bimserverPort) :
+    QObject(parent),
+    m_networkManager(new QNetworkAccessManager)  
+  {
+    QString bimserver = "http://" + bimserverAddr + ":" + bimserverPort + "/json";
+    m_bimserverURL = QUrl(bimserver);
+    qDebug() << "BC initialized";
```

Comment:
Sorry about that. That was for my debugging purposes and I deleted the qDebug() lines.